prow/github/helpers.go

```diff
@@ -82,3 +82,25 @@ func PermissionsFromLevel(permission RepoPermissionLevel) RepoPermissions {
 		return RepoPermissions{}
 	}
 }
+
+func MatchCardToContent(cardURL, contentURL string) bool {
+	if strings.Compare(cardURL, contentURL) == 0 {
+		return true
+	}
+
+	properContent := strings.Replace(cardURL, "repos/", "", 1)
```

Comment:
Removing `repos/` and `api.` works but it's very much unclear what is actually going on here. My opinion would be that we could instead:

 - parse the URL
 - split the path on slashes
 - assert structure to the parts

Then we can compare org, repo and issue ID from both? 

Either way works, the current impl is just not very clear as to what is going on

Reply:
given a URL it is not possible to predict if the url has org or  org/repo. I am not sure if parsing gets us more predictability. I feel we should iterate once the functionality is working at least.

Reply:
Right, if you can't tell that the parts of the path once you split them contain an org and repo, return false (if that is the API you want). I would suggest returning an error if you cannot validate that the input URLs are exactly as you expect them. I am not a fan of the current approach since it's really not clear why it works